pkg/scale/varying_data_type.go

```diff
@@ -66,8 +66,11 @@ func (vdt *VaryingDataType) Set(value VaryingDataTypeValue) (err error) {
 }
 
 // Value returns value stored in vdt
-func (vdt *VaryingDataType) Value() VaryingDataTypeValue {
-	return vdt.value
+func (vdt *VaryingDataType) Value() (VaryingDataTypeValue, error) {
+	if vdt.value == nil {
+		return nil, fmt.Errorf("vdt value is nil")
```

Comment:
```suggestion
		return nil, fmt.Errorf("VaryingDataTypeValue has not been set")
```
See Ed's PR #2708, he defines sentinel errors for the scale package.  This should be a sentinel error as well.  Possibly a public one IMO.